### PR TITLE
Addition endpoints for toolbox

### PR DESCRIPTION
**Pull request by joyce-wu**

### CODE REVIEW

**Review by gabrielsaruhashi — COMMENTED** [review]
hey @joyce-wu ! there are a couple of merge conflicts on this PR! could you check those out?

**Review by gabrielsaruhashi — CHANGES_REQUESTED** [review]
hey @joyce-wu there are a couple of merge conflicts that weren't resolved / large chunks of commented code. Could you take a look at those?

File: app/Jobs/SendNewsletter.php [hunk 1]
```diff
@@ -81,7 +81,11 @@ public function handle()
         $this->newsletter->save();
 
         foreach ($contacts as $contact) {
+<<<<<<< HEAD
+            usleep(500000);
```
Comment (gabrielsaruhashi) [inline app/Jobs/SendNewsletter.php:85]: hey! there's still one conflict

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -349,7 +349,7 @@ public function get_referrals(Request $request, $id) {
 
         $referrals = User::where('referred_by', $u->referral_link)->get();
         $referrals_ids = $referrals->pluck('id')->toArray();
-        
+
```
Comment (gabrielsaruhashi) [inline app/Http/Controllers/UserController.php:352]: nit: wanna delete this extra space?

File: app/Http/Controllers/DesignCategoryController.php [hunk 3]
```diff
@@ -52,13 +52,49 @@ public function __construct() {
...
+          $category = DesignCategory::find($id);
+
+          if (!$category) {
+            return api_response(404, "ERROR", "Invalid Category ID", []);
+          }
+          return api_response(200, "OK", "", $category);
+      }
+
+      $categories = DesignCategory::all();
+      $total = $categories->count();
```
Comment (gabrielsaruhashi) [inline app/Http/Controllers/DesignCategoryController.php:84]: any way we can move this content range logic to a util file? It seems to be duplicated across all GET functions

File: app/Http/Controllers/AuthController.php [hunk 4]
```diff
@@ -96,6 +96,60 @@ public function login_with_token(Request $request) {
       return api_response(400, "ERROR", "Missing Fields", $errors);
     }
 
+    public function login_admin(Request $request) {
+      $data = json_decode($request->getContent(), true);
```
Comment (gabrielsaruhashi) [inline app/Http/Controllers/AuthController.php:100]: what's the difference between this function and the plain login()
Couldnt we just use that?

Also, I think we're missing the  some layer of validation ot ensure the user is an Admin (and return 404 if not)

**Review by gabrielsaruhashi — CHANGES_REQUESTED** [review]
super cool! wrote a couple of comments 🔧

**Review by gabrielsaruhashi — APPROVED** [review]
🔥

**Review by jessehorne — APPROVED** [review]

**Pull request merged**